ivy/functional/backends/torch/statistical.py

```diff
@@ -83,6 +83,10 @@ def mean(
             return ivy.inplace_update(out, x)
         else:
             return x
+    if "float" not in str(x.dtype) and "complex" not in str(x.dtype):  # unsupported
+        return torch.mean(x.to(torch.float32), dim=axis, keepdim=keepdims, out=out).to(
```

Comment:
Same here, we shouldn't really hardcode type casting to `float32` on this line, it should be based on whatever's the value of `x.dtype`